File: c++代码/1557. 可以到达所有点的最少点数目.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>> &edges)
    {
        vector<int> bian(n);
        for (int i=0;i<edges.size();i++)
        {
            bian[edges[i][1]] ++;
        }
        vector<int> ans;
        for (int i = 0;i<bian.size();i++)
        {
            if (bian[i] == 0) ans.push_back(i);
        }
        return ans;
    }
};

File: 力扣/图/1557. 可以到达所有点的最少点数目.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: list) -> list:
        indeg = [0] * n
        for item in edges:
            # 记录入读节点个数
            indeg[item[1]] += 1
        
        q = [u for u in range(n) if indeg[u] == 0]
        return q